.github/workflows/image-publish.yml

```diff
@@ -64,10 +74,30 @@ jobs:
       - name: Docker Build and Push
         run: |
           SHORT_SHA1=$(git rev-parse --short=7 HEAD)
-          docker buildx build --platform linux/amd64 -f build/dockerfiles/assembly.Dockerfile --push -t quay.io/che-incubator/che-code:insiders -t quay.io/che-incubator/che-code:next -t quay.io/che-incubator/che-code:insiders-${SHORT_SHA1} .
+          if [[ ${{ env.BRANCH_NAME }} == main ]]; then
```

Comment:
You set `env.BRANCH_NAME` to "insiders" on line 51, so this check will never pass.